### ISSUE TITLE
Bug: Unable to leave request space access after leaving it

### ISSUE THREAD

**Issue by raphj** [tracim/tracim#3738]
## Description and expectations

After leaving an on request space, the accepted subscription is still there, seemingly preventing a new request.

**Comment by inkhey** [comment]
@Skylsmoi Just checked a bit the code, it look like accepted subscription stay as they are not deleted, but, there are no check that unallow to re-requestion an access (using the put endpoint).

**Comment by Skylsmoi** [comment]
Yes, there isn't exactly an issue with the code, it's about the meaning of the api responses.
The question is, afaik, whether the backend should delete the accepted requests once the user leaves the spaces or whether the frontend should implement the added intelligence of checking that an accepted request is part of the currently available spaces of the user

**Comment by lebouquetin** [comment]
Is there a unique key on user/workspace request? on a functionnal point of view, there is no reason for not allowing a user to re-request to join a space.

Making a PUT if a request is already there is counter-intuitive, I think. I would expect:

- allow to POST a new request
- force to DEL the existing request before POST a new
- POST a `renew` request

The choice is not critical, I think, but remember that the API should be as auto-explain as possible (for example, @raphj implemented kanban the wrong way because of both wrong property naming and deficient documentation)

**Comment by Skylsmoi** [comment]
Do your `renew` means a specific status for a request ?
I don't think so since you propose to delete it before recreating it.

Anyways, I think this process is already possible.
The issue is that the frontend displays "Request accepted", since the api returns the request as "accepted" even if the user as left the space.
So the user can't see the button to create a new request.

**Comment by grignards** [comment]
What about deleting the related request when a user leaves a space? So that the situation comes back to a new join?
This seems to be the easier implementation provided we don't have/see a need to keep all join requests.
